Correct the FCGI installation instructions

Edouard-chin opened the pull request:
Fix the readme:

- I'm taking over #1102 since it has been open for a long time now (Thanks @yskn for opening the issue!)
- http://www.fastcgi.com is no longer available, this replaces the instruction to download the tar from a github archive
- Closes #1102

cc/ @rafaelfranca

Edouard-chin commented:
Looks like builds are broken on ruby-head since at least revision [62380](https://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/62380) (issues with bundler 1.16.1)

Not sure why build are failing on jruby though

matthewd commented:
> `BodyProxy#each` should return an enumerator when no block is passed

Should it?
 https://github.com/rack/rack/blob/d1363a66ab21738d53bf13cece3652ada0f01976/SPEC#L245-L246

Edouard-chin commented:
Hmm indeed, didn't read the SPEC. So maybe it's [ActionDispatch::Response::RackBody](https://github.com/rails/rails/blob/20c91119903f70eb19aed33fe78417789dbf070f/actionpack/lib/action_dispatch/http/response.rb#L124) that needs to be fixed.

I opened the PR because I saw an inconsistency between both

rafaelfranca commented:
It is open for interpretations. If `each` return a enumerator, it still respond to each and only yield string values.

But most of the body objects don't return an enumerator, except for Rails' `ActionDispatch::Response::Buffer`.

Edouard-chin commented:
Doing a spring clean of my open PR, I have removed the code to `BodyProxy#each` since it's open for interpretation and we fixed the code in our own app at some point instead.

This PR now just corrects the installation instruction which is currently broken
